design/design-doc-template-stacks-experience.md

```diff
@@ -0,0 +1,473 @@
+      properties:
+        ...
+        spec:
+          type: object
+          properties:
+            ...
+            image:
+              type: string
+              description: A custom wordpress container image id to use
+              # Defaults are specified like this, using the schema validation for CRD fields.
+              # For more about how this works with CRDs, see:
+              # https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#defaulting
+              default: "wordpress:4.6.1-apache"
+```
+
+For the full example, see the sample in the
+[template stack experience repository](https://github.com/suskin/template-stack-experience/tree/master/wordpress-workload/go-templates/default-variables).
+
+### How processing objects works under the covers
+
+The imagined implementation for how objects sent to the stack for
+rendering become resources is that the object's fields become the input
+for the template rendered by the controller. For more details, see the
+[detailed examples](https://github.com/suskin/template-stack-experience)
+or the [design document about the
+internals](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-template-stacks.md).
+
+### Templating/configuration engine
+
+Template stacks will not be opinionated about which templating engine is
+used. We plan to support multiple configuration engines. The engine will
+be configurable by setting values in the `stack.yaml`.
+
+### Lifecycle hooks
+
+We expect to eventually support lifecycle hooks. See the [speculative
+design in the template stacks experience repo](https://github.com/suskin/template-stack-experience/tree/master/wordpress-workload/go-templates/lifecycle-hook)
+for more details about what that could look like.
+
+### Internal representation of templates
+
+See the [design doc on the internals of the Template Stack
+implementation](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-template-stacks.md).
+
+### The template stack controller
+
+See the [design doc on the internals of the Template Stack
+implementation](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-template-stacks.md).
+
+## Example use-cases
+
+See the [Template Stack
+Experience](https://github.com/suskin/template-stack-experience) code
+examples for some realistic examples of how different use-cases work
+with a realistic workload.
+
+Examples covered by the repository all currently use go templates as the
+templating engine. The following cases are covered by the examples:
+
+* No variables at all in a set of configurations.
+* Variables in configurations with default variable values.
+* Variables in configurations, with values specified by users when a
+  render is requested.
+* Resource packs.
+* Multiple CRDs defined in a single Stack.
+* Multiple rendered variants of a given configuration.
+* Composing configuration using multiple Stacks.
+
+### Helm charts
```

Comment:
this is perhaps where we can get into more depth around how helm charts could be reused and what the experience looks like.  it would be **great** to see that in the https://github.com/suskin/template-stack-experience repo too, with basically a concrete example of what we expect a fully building repo that reuses a helm chart in a stack to look like.

Reply:
I added a few links to this section to some more detailed samples, so give readers a place to see more details.